staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -771,7 +778,7 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 // received. It handles retry behavior and up front validation of requests. It will invoke
 // fn at most once. It will return an error if a problem occurred prior to connecting to the
 // server - the provided function is responsible for handling server errors.
-func (r *Request) request(fn func(*http.Request, *http.Response)) error {
+func (r *Request) request(fn func(*http.Request, *http.Response), span opentracing.Span) error {
```

Comment:
there's probably a way to do this without passing the span explicitly, instead using the Context to contain the span.